tests/tracking/default_experiment/test_registry.py

```diff
@@ -0,0 +1,169 @@
+from importlib import reload
+from unittest import mock
+import pytest
+
+import mlflow.tracking.default_experiment.registry
+from mlflow.tracking.default_experiment.databricks_notebook_experiment_provider import (
+    DatabricksNotebookExperimentProvider,
+)
+from mlflow.tracking.default_experiment.databricks_job_experiment_provider import (
+    DatabricksJobExperimentProvider,
+)
+from mlflow.tracking.default_experiment.registry import (
+    DefaultExperimentProviderRegistry,
+    get_experiment_id,
+)
+
+# pylint: disable=unused-argument
+
+
+def test_default_experiment_provider_registry_register():
+    provider_class = mock.Mock()
+
+    registry = DefaultExperimentProviderRegistry()
+    registry.register(provider_class)
+
+    assert set(registry) == {provider_class.return_value}
+
+
+def test_default_experiment_provider_registry_register_entrypoints():
+    provider_class = mock.Mock()
+    mock_entrypoint = mock.Mock()
+    mock_entrypoint.load.return_value = provider_class
+
+    with mock.patch(
+        "entrypoints.get_group_all", return_value=[mock_entrypoint]
+    ) as mock_get_group_all:
+        registry = DefaultExperimentProviderRegistry()
+        registry.register_entrypoints()
+
+    assert set(registry) == {provider_class.return_value}
+    mock_entrypoint.load.assert_called_once_with()
+    mock_get_group_all.assert_called_once_with("mlflow.default_experiment_provider")
+
```

Comment:
Can we also add a test on testing the ordering of provider registration is honored correctly? That means when both `DatabricksNotebookExperimentProvider` and `DatabricksJobExperimentProvider` are in context, `DatabricksJobExperimentProvider` is returned.

Reply:
There should already be a test for that here: https://github.com/mlflow/mlflow/pull/5309/files#diff-33f89924b0ce3f56c6d3e332907629a440d69b39e3c8e9c8dd3aff073d2b7d20R144
Let me know if I am missing something